src/common/common.go

```diff
@@ -248,35 +248,85 @@ func parseOpenshiftVersionIntoMajorMinorZOnly(version string) string {
 }
 
 // HasValidvSphereCredentials returns true if the the platform is
-// vSphere and the install config overrides contains real
+// vSphere and the install config or equivalent overrides contains real
 // credential values and not placeholder values.
 // Deprecated credential fields are not considered valid.
-func HasValidvSphereCredentials(ctx context.Context, ic inventory_client.InventoryClient, log logrus.FieldLogger) bool {
+func HasValidvSphereCredentials(ctx context.Context, ic inventory_client.InventoryClient, kc k8s_client.K8SClient, log logrus.FieldLogger) bool {
 	cluster, callErr := ic.GetCluster(ctx, false)
+	if cluster != nil {
```

Comment:
I see that invoker check is part of another function so i believe it should be fine, right?

Reply:
In additional to having valid credentials, RemoveUninitializedTaint has additional checks on cluster version, invoker, and platformType to determine if the taints should be removed. That seemed to be a specific ABI case, but the function felt that the appropriate place for it. If we don't run RemoveUninitializedTaint, then the additional checks would need to be elsewhere.